src/Entity.php

```diff
@@ -3986,4 +3993,55 @@ public static function badgeCompletenameById(int $entity_id): ?string
         }
         return null;
     }
+
+    /**
+     * Return HTML code for entity badge showing its completename with last entity as HTML link.
+     *
+     * @param object $entity
+     *
+     * @return string|null
+     */
+    public static function badgeCompletenameLink(object $entity): string
+    {
+        // `completename` is expected to be received as it is stored in DB,
+        // meaning that `>` separator is not encoded, but `<`, `>` and `&` from self or parent names are encoded.
+        $names     = explode(' > ', trim($entity->fields['completename']));
+        $last_name = Sanitizer::decodeHtmlSpecialChars(array_pop($names));
```

Comment:
Decoding process can be mutualized.
```suggestion
        // Convert the whole completename into decoded HTML.
        foreach ($names as &$name) {
            $name = Sanitizer::decodeHtmlSpecialChars($name);
        }

        $last_name = array_pop($names);
```